src/main/java/wayoftime/bloodmagic/tile/TileSoulForge.java

```diff
@@ -131,7 +131,17 @@ public void tick()
 			}
 		}
 
-		double soulsInGem = getWill(EnumDemonWillType.DEFAULT);
+		double soulsInGem = 0;
+		EnumDemonWillType typeInGem = EnumDemonWillType.DEFAULT;
+		for (EnumDemonWillType type : EnumDemonWillType.values())
+		{
+			soulsInGem = getWill(type);
+			if (soulsInGem > 0)
```

Comment:
Should perform the check versus the minimum amount of will in the recipe, in the case of there being a gem with multiple aspects.

Reply:
A gem with multiple aspects? I'd never even considered that as a possibility... But it sounds like it'd be pretty cool.

Reply:
Adjusted by the latest commit.